src/service/action-impl.js

```diff
@@ -332,6 +334,16 @@ export class ActionService {
         const element = dev().assertElement(event.target);
         this.trigger(element, name, event, ActionTrust.HIGH);
       });
+    } else if (name == 'focus') {
+      this.root_.addEventListener(name, event => {
+        const element = dev().assertElement(event.target);
+        this.trigger(element, name, event, ActionTrust.HIGH);
```

Comment:
There is one tricky thing we need to figure out:
`focus`/ `blue` can happen on elements without user interaction in certain cases such as using '#idOfInput` in Url or using `autofocus` attribute . We should either not fire the events in those cases or fire it with "Low-trust" (I vote for former).

I am not sure how we can detect it, I believe the `event` target when focus is done via user-action would have something different than when it is automatic. Some investigation is needed.

Reply:
Pinging @dvoytenko to take a look here.